sdk/client.go

```diff
@@ -5,14 +5,19 @@ import (
 	"fmt"
 	"net/http"
 
+	healthcheck "github.com/ONSdigital/dp-api-clients-go/v2/health"
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
 	dprequest "github.com/ONSdigital/dp-net/v2/request"
 	"github.com/ONSdigital/dp-search-reindex-api/models"
 )
 
 //go:generate moq -out ./mocks/client.go -pkg mocks . Client
 
 type Client interface {
+	Checker(ctx context.Context, check *health.CheckState) error
+	HealthClient() *healthcheck.Client
```

Comment:
(suggestion): I think the interface ```HealthClient``` kind of cause stutter. ```https://go.dev/doc/effective_go```. An effective name would be something like Health(), so the consumers can call ```client.Health()```. rather than ```client.HealthClient```.